apps/emqx/src/emqx_router_helper.erl

```diff
@@ -101,6 +101,12 @@ monitor(Node) when is_atom(Node) ->
 
 init([]) ->
     process_flag(trap_exit, true),
+    %% Make sure no other node can do late routes cleanup when this node is up.
+    true = global:set_lock({?LOCK(node()), self()}),
```

Comment:
Good idea! If we do `global:set_lock({?LOCK(node()), self()}, [node()])` here, I wonder will it be as safe as it is now? Otherwise there's a risk of blocking the node startup if some replicant somewhere is unreachable. Or maybe I'm missing something. 🤔

Reply:
Honestly, that's was suggested by @ieQu1, so not my idea. :slightly_smiling_face: 
You mean that this node tries to acquire the lock on all currently live nodes (`[node()|nodes()]`) but some of these nodes are unreachable? If so, `global` must definitely handle it by ignoring unreachable nodes:
```
set_lock_on_nodes(Id, Nodes) ->
    case local_lock_check(Id, Nodes) of 
        true ->
            Msg = {set_lock, Id},
            {Replies, _} = 
                gen_server:multi_call(Nodes, global_name_server, Msg),
```
Here it ignores BadNodes, so if a node is unreachable, the lock will be acquired only on live nodes.

Reply:
Yes, but it does so with `infinity` timeout (according to `gen_server:multicall/3`). Granted it won't realistically become stuck, but I must admit it's not really clear how long the multicall itself will take in the worst case (i.e. when some node gone down but this node haven't noticed that yet).

> If we do `global:set_lock({?LOCK(node()), self()}, [node()])` here

Anyway, disregard that, it's unsafe, haven't thought it through enough. 🫠

Reply:
However, I don't honestly see an alternative that is simple enough, so maybe just keep that in mind / leave a comment here? Admittedly, this situation shouldn't really happen often.

Reply:
I think the worst case (if a node is really unresponsive) depends on the `ticktime`, which, acc. to the docs is 
```
MaxT = NetTickTime + NetTickTime / NetTickIntensity
``` 
(https://www.erlang.org/doc/man/kernel_app#net_ticktime)
So, even though gen_server multicall uses infinity timeout, it's expected to receive 'DOWN' msg if the node is not responding..

Reply:
Added a comment